### ISSUE TITLE
%1 is not a valid win32 application

### ISSUE THREAD

**Issue by KyNicholas** [Purfview/whisper-standalone-win#46]
I hope you're doing well. I wanted to share an issue I'm facing and seek your advice. I've been using Whisper-Faster for testing on various recordings, and it has been working fine until today when I encountered an error after a Windows update. The error message says, "%1 is not a valid Win32 application." I tried searching for a solution on Google, but the results returned various variations of the problem, and I'm unsure where to begin troubleshooting.

All the required DLLs are included in the same folder as the application.

If you have any suggestions or insights on how to resolve this issue, I would greatly appreciate your help.
Thank you in advance!

![1](https://github.com/Purfview/whisper-standalone-win/assets/134874930/204c7ed0-fdb4-46d7-b221-f507ba0cbe77)

**Comment by Purfview** [comment]
Because your OS is 32-bit, you need 64-bit to run it.

**Comment by KyNicholas** [comment]
Actually, I use Windows 11 64-bit.
I have attached the picture.


![2](https://github.com/Purfview/whisper-standalone-win/assets/134874930/19aef26b-e4a4-4c59-96c9-4cf2a783e5fe)

**Comment by Purfview** [comment]
You should ask Windows support why after the update it want x64 DLL to be x86 in x64 Windows.

**Comment by Purfview** [comment]
Use this guide to uninstall problematic update: https://pureinfotech.com/uninstall-update-windows-11

**Comment by KyNicholas** [comment]
Actually, I found out that the problem is not related to the Windows update. I tried after performing a fresh Windows installation. Upon investigating the Event Viewer, it became evident that the issue is associated with the "ucrtbase.dll" file.


![3](https://github.com/Purfview/whisper-standalone-win/assets/134874930/fc1f83c7-0784-4f3b-8123-0e7b0d9a0db8)

**Comment by Purfview** [comment]
>  I found out that the problem is not related to the Windows update. I tried after performing a fresh Windows installation.

I don't see that it's not related, then related to what? Are you sure your new installation is exactly same  [with same updates]  when program was working?

I think that "ucrtbase.dll" is from Microsoft Visual C++ Redistributable, maybe you updated this one?

Or maybe you are running it in some different way. Do you run it as Administrator? You need to remember what changed before it started to crash.

**Comment by emcodem** [comment]
@KyNicholas 
When you execute the app, first thing is all python scripts and loads of dll's are extracted to %temp%\_MEIxxxx - this _MEI folder is usually deleted at program end but maybe in your case you have lots of them still there because the app crashed?
%temp% is for me: C:\Users\emcod\AppData\Local\Temp

You can in theory use tools like dependencies GUI to check the dll's in this folder and see if all dependencies of all dll's can be loaded. It's a pretty advanced topic to debug app dependencies this way but there are lots of articles about it. 

Unfortunately the error "is not a valid win32 application" can mean so much things, i even get it when trying to load an "empty" dll file into my c++ applications.

**Comment by Purfview** [comment]
Yeah, that error can be anything. I remember getting it ages ago when my RAM or HDD was failing.

I doubt that it's because of remnants of "_MEIxxxxx" folder, every time it's randomly named, even if a name coincides then it would overwrite existing files in it.

Actual crash happens in PyAV lib on this line `from av._core import time_base, library_versions`.
I guess "av._core " is `_core.cp38-win_amd64.pyd` file.

**Comment by emcodem** [comment]
> I doubt that it's because of remnants of "_MEIxxxxx" folder
Sorry if what i wrote points in that direction. The only reason why i point to the MEI folder is that this is where the actual application files can be found and investigated. 
I don't have any assumption or suspicion about the origin of this failure and i guess the OP is not interested in doing deep investigations and gave up - but i thought it might be interesting for future readers to know a little backgrounds about how the stuff works and can be debugged. 

Btw, as we know the error comes from av module, the dll that causes the issue is most likely found in _MEIxxx\av.libs folder. Knowing this, it would also be worth to start Process Monitor and filter for Process Name whisper-faster.exe, then start the process and see which dll from the av.libs folder was the last one being accessed. 
ucrtbase.dll is just the C runtime, we actually have a copy of it in the _MEI folder, so i would assume that ONLY this one is used. In procmon i see that both are used, the one from MEI folder AND the one from system32 which is kind of strange but ok so be it...

**Comment by Purfview** [comment]
I would start looking with actual Python with `av._core import time_base, library_versions`.

**Comment by KyNicholas** [comment]
I tried many things to fix the problem, but none of them worked. Here's what I did:

* I tried reinstalling Microsoft Visual C++ Redistributable packages.
* I attempted to stop Windows updates using the firewall.
* I installed Python on my computer.
* I used a tool called pyinstxtractor-ng to extract a file, changed a part inside it, and tried running it manually.
* I also downloaded ctranslate2 and tried using it, but it gave me the same error.

Even after all these attempts, the error was still there even after 2 hours. Just so you know, the computer I used for audio transcription had a program called Deep Freeze installed since the very beginning, and it was never changed or unfrozen.

In the end, instead of dealing with more Windows issues, I decided to do something simpler. I reinstalled *Windows 11* using the *latest version ISO* I downloaded, and that fixed the problem.
